client/src/pages/Home/Home.tsx

```diff
@@ -1,23 +1,35 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { Container, DownArrow, PageTitle, UpArrow } from "./Home.styles";
-import { BudgetCardList } from "../../components";
+import { BudgetCardList, Loading } from "../../components";
 import { useQuery } from "react-query";
 import { getAllBudgets } from "../../API/BudgetMethods";
 import { IoIosArrowUp, IoIosArrowDown } from "react-icons/io";
 
-const Home = () => {
-  const [position, setPosition] = useState<number>(12);
+interface IHome {
+  displayLoader: boolean;
+  setDisplayLoader: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const Home: React.FC<IHome> = ({ displayLoader, setDisplayLoader }) => {
   const fetchAllBudgets = async () => {
     const data = await getAllBudgets();
     return data;
   };
-
   const { data, isLoading, status, refetch } = useQuery(
     "budgets",
     fetchAllBudgets,
     {}
   );
 
+  const [position, setPosition] = useState<number>(12);
+
+  useEffect(() => {
+    displayLoader &&
+      setTimeout(() => {
+        setDisplayLoader(false);
+      }, 3000);
```

Comment:
You may need to add a cleanup function here.

Reply:
Never done this before haha but I changed it - let me know if it's okay now